store.go

```diff
@@ -561,6 +575,12 @@ type Store interface {
 	// usually by deleting layers and images which are damaged.  If the
 	// right options are set, it will remove containers as well.
 	Repair(report CheckReport, options *RepairOptions) []error
+
+	// List returns a ListResult structure that contains layer, image, or container
+	// extracts according to the values in ListOptions.
+	// List method uses a locking mechanism so that the result is the current snapshot
+	// of the items listed.
```

Comment:
Callers don’t care that this is implemented with locks (and we don’t want to promise that), just that the values are consistent.